src/pages/EditorPage.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from "react";
+import { useNavigate } from "react-router";
+import { createDataSamplesPageUrl } from "../urls";
+import { useStore } from "../store";
+
+const EditorPage = () => {
+  const navigate = useNavigate();
+  const model = useStore((s) => s.model);
+  const setEditorOpen = useStore((s) => s.setEditorOpen);
+
+  useEffect(() => {
+    if (!model) {
+      navigate(createDataSamplesPageUrl());
+    } else {
+      setEditorOpen(true);
+    }
+
+    return () => {
+      if (model) {
+        setEditorOpen(false);
+      }
+    };
+  }, [model, navigate, setEditorOpen]);
```

Comment:
Not a fan of this but it will catch using the browser forward/back buttons, and navigation by URL / link.

Reply:
As discussed `setEditorOpen(false)` now handles the model condition internally.